room.cpp

```diff
@@ -875,6 +877,69 @@ void Room::Private::updateDisplayname()
         emit q->displaynameChanged(q);
 }
 
+void Room::Private::toJson(QJsonObject &out) {
+    QJsonValue nowTimestamp { QDateTime::currentMSecsSinceEpoch() };
+    QJsonArray stateEvents;
+
+    QJsonObject nameEvent;
+    nameEvent["type"] = QString("m.room.name");
+    QJsonObject nameEventContent;
+    nameEventContent["name"] = this->name;
+    nameEvent["content"] = nameEventContent;
+    stateEvents.append(QJsonValue(nameEvent));
```

Comment:
Don't try so hard constructing `QJsonValue`s explicitly. Those constructors will be inserted by the compiler.